tests/end_to_end/andor_return/irix-o2-out.c

```diff
@@ -0,0 +1,9 @@
+s32 test(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
+    if ((arg0 != 0) || (arg1 != 0)) {
+        if ((arg2 != 0) || (arg3 != 0)) {
+            return arg0 + arg1;
+        }
+        return arg2 + arg3;
+    }
+    return arg2 + arg3;
```

Comment:
A TODO sounds good! I think the existing duplicate return node emission might have been an earlier attempt at doing what `duplicate_premature_returns` does now? Or just a bug.

Reply:
Cool, I left the TODO in `emit_node` so this can be revisited: https://github.com/matt-kempster/mips_to_c/pull/114/files#diff-fe6eb70d8a821eb6a7b0aa6194f261ff29d461f0bca93ef30bfb32b061f381e2R281-R282